src/protocol/resp/src/parse/command.rs

```diff
@@ -0,0 +1,261 @@
+// Copyright 2023 Pelikan Foundation LLC.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use std::borrow::Cow;
+use std::str::FromStr;
+
+use crate::message::Message;
+
+use super::{ArrayParser, ParseError, ParseResult, Parser};
+
+enum Detail<'a, 'p> {
+    Inline(Vec<Cow<'a, [u8]>>),
+    Resp(ArrayParser<'a, 'p>),
+}
+
+pub struct CommandParser<'a, 'p> {
+    detail: Detail<'a, 'p>,
+}
+
+impl<'a, 'p> CommandParser<'a, 'p> {
+    pub fn new(parser: &'p mut Parser<'a>) -> ParseResult<'a, Self> {
+        match parser.peek() {
+            Some(b'+' | b'-' | b':' | b'$' | b'*') => Ok(Self {
+                detail: Detail::Resp(parser.parse_array()?.ok_or(ParseError::Custom {
+                    expected: "a non-nil array",
+                    found: "a nil array",
```

Comment:
nit: The RESP spec uses the term `Null array` and `Null bulk string`, both of which map to `nil` or `NULL` in common programming languages, but it is never called a `nil array`.